test/tests/dom/shadowDOMFunctions.js

```diff
@@ -0,0 +1,209 @@
+const customWebWrapper = class extends HTMLElement {
+    constructor() {
+        super();
+        const shadowRoot = this.attachShadow({ mode: 'open' });
+        const shadowDOMContainer = document.createElement('div');
+        shadowDOMContainer.setAttribute('id', 'inner-host-div');
+        shadowRoot.appendChild(shadowDOMContainer);
+    }
+};
+
+const customWebComponent = class extends HTMLElement {
+    constructor() {
+        super();
+        const shadowHost = this.attachShadow({ mode: 'open' });
+        const shadowDOMContainer = document.createElement('div');
+        const testSpan = document.createElement('span');
+        testSpan.setAttribute('id', 'inner-span');
+        testSpan.textContent = 'text from custom web component';
+        // Append it to the shadow root
+        shadowDOMContainer.appendChild(testSpan);
+        shadowHost.appendChild(shadowDOMContainer);
+    }
+};
+
+customElements.define('custom-web-component', customWebComponent);
+customElements.define('custom-wrapper', customWebWrapper);
+
+describe('Web components', () => {
+
+    beforeEach(() => {
+        if (!document?.body) {
+            throw new Error('Body not found');
+        }
+        
+        const customElement = document.createElement('custom-web-component');
+
+        // Clean the DOM
+        if (document.body) {
+            document.body.innerHTML = '';
+            document.body.appendChild(customElement);
+        }
+
+    
+        if (!customElement || !customElement.shadowRoot) {
+            throw new Error('custom element does not have shadow root');
+        }
+    
+    });
+
+    describe('isShadowElement cases', () => {
+  
+        it('should return true if parent node is shadow root',  () => {
+            const innerElement = document.querySelector('custom-web-component')?.shadowRoot?.querySelector('#inner-span');
+    
+            if (!innerElement) {
+                throw new Error('unable to find inner element');
+            }
+            
+            const result = isShadowElement(innerElement);
+    
+            if (!result) {
+                throw new Error(`Expected result to be true, got ${ String(result) }`);
+            }
+        });
+    
+        it('should return false if parent node is not shadow root', () => {
+            const testElement = document.createElement('div');
+            const result = isShadowElement(testElement);
+    
+            if (result) {
+                throw new Error(`Expected result to be false, got ${ String(result) }`);
+            }
+        });
+    });
+    
+    describe('getShadowRoot cases', () => {
+        it('should return shadow root host', () => {
+            const innerElement = document.querySelector('custom-web-component')?.shadowRoot?.querySelector('#inner-span');
+    
+            if (!innerElement) {
+                throw new Error('unable to find inner element');
+            }
+            const result = getShadowRoot(innerElement);
```

Comment:
@gabrielo91  The test says it should return shadow root host but we are just getting the root.  Is this correct?

Reply:
My mistake, the test declaration should be `should return shadow root`. Fixed